harness/determined/pytorch/deepspeed/_deepspeed_trial.py

```diff
@@ -291,6 +298,22 @@ def cleanup_iterator() -> None:
                 self._run()
 
     def _run(self) -> None:
+        # Special code path only used for DeepSpeed Autotuning.
+        if self._dsat_mode:
+            ops = self.context._core.searcher.operations()
+            op = next(ops)
+            # TODO: remove `while True` in favor of exact number of steps which should run.
+            # TODO: also assuming here that op.length is always the number of *steps* to train for
```

Comment:
Do these TODOs have tickets assigned with them?  Or are they the sort of "TODO: some day I'll fix the whole world, but I don't plan on doing it any time soon" kind of TODOs?  (honestly there's reasons for each, I'm just curious which kind these are)

Reply:
I don't believe these have tickets and I __think__ the second one is resolved and can be removed? @garrett361 I believe the first one is also potentially a non-issue that should be resolved with the work to remove the workload_sequencer. Perhaps it would be best to simply remove these comments

Reply:
Ah oops, looked for `TODO`s in `/dsat`, but forgot to look here, good catch.

I believe that both can be removed and the `while True` can be replaced with `for step in op.length`, but obviously need to verify.  

Both `TODO`s were todo-before-merging, though neither ticketed.

Reply:
Removed in #6988

Reply:
side note, I recommend using `XXX:` or `FIXME:` or `TODO(premerge)` or really anything other than plain `TODO` to annotate fix-before-merging type things.  There are nearly zero instances of either of those in the entire codebase, so you can `git grep XXX` and easily identify the things you wanted to revisit.

I do this _all_ the time, and it really relieves a lot of mental stress because by liberally annotating things I know aren't fully baked, I can focus on one thing at a time without dropping details.

Reply:
That sounds like a better process, will try to remember in the future